frontend/src/routes/AdminPageAbakusLeaderView/index.js

```diff
@@ -91,132 +88,131 @@ class AdminPage extends Component {
       .then((data) => {
         data.map(({ url, jsonData }) => {
           if (url.includes("/application/")) {
-            this.setState({
+            setState({
               applications: jsonData,
             });
           } else if (url.includes("/admission/")) {
-            this.setState({
+            setState({
               admission: jsonData[0],
             });
           } else if (url.includes("/group/")) {
-            this.setState({
+            setState({
               groups: jsonData,
             });
           }
         });
-        this.setState({
+        setState({
           isFetching: false,
         });
       })
       .catch((error) => {
-        this.setState({ error, isFetching: false });
+        setState({ error, isFetching: false });
       });
 
-    this.setState({
+    setState({
       user: { name: djangoData.user && djangoData.user.full_name },
     });
```

Comment:
Have you tested this? I think these `setState`s might delete the rest of the state. Class component `setState` would merge the new state into the old state, but I'm pretty sure these `useState`-`setState` methods will just overwrite it.

Splitting the state into multiple `useStates` would avoid the problem.

Reply:
Ah good catch, didn't notice that I used this in that file as well - will fix. Weirdly enough it seemed to work fine when tested but probably not thoroughly enough